src/west/main.py

```diff
@@ -286,31 +296,35 @@ def run_command(self, argv):
                                 list(malformed.args)))
 
     def run_builtin(self, args, unknown):
-        if self.mve:
+        if self.mle:
+            self.handle_builtin_manifest_load_err(args)
+
+        cmd = self.builtins.get(args.command, self.builtins['help'])
+        cmd.run(args, unknown, self.topdir, manifest=self.manifest)
+
+    def handle_builtin_manifest_load_err(self, args):
+        if isinstance(self.mle, ManifestVersionError):
             if args.command == 'help':
-                log.wrn(mve_msg(self.mve, suggest_ugprade=False) +
+                log.wrn(mve_msg(self.mle, suggest_ugprade=False) +
                         '\n  Cannot get extension command help, ' +
                         "and most commands won't run." +
                         '\n  To silence this warning, upgrade west.')
             elif args.command in ['config', 'topdir']:
                 # config and topdir are safe to run, but let's
                 # warn the user that most other commands won't be.
-                log.wrn(mve_msg(self.mve, suggest_ugprade=False) +
+                log.wrn(mve_msg(self.mle, suggest_ugprade=False) +
                         "\n  This should work, but most commands won't." +
                         '\n  To silence this warning, upgrade west.')
```

Comment:
Is this working as expected ?
When I run:
```
$ west topdir
/tmp/import_test
```
there is no warning.

but on other commands, I do get the help text, as example:
```
$ west status
FATAL ERROR: some imported manifest data is missing. To fix, run:
  west update
```

btw. I assume the hint is wrong. Why would updating `west` help silence the warning ?

Reply:
Hint should be fixed, see handle_builtin_manifest_load_err. Thanks.

It's fine to run `west topdir` without a manifest, so the success there is expected. For `topdir` to succeed, a `.west` must be found, but that's it. On the other hand, `west status` really needs a manifest to do its job.

Part of these changes are deliberately splitting up the built-in command error handling depending on whether the commands actually need the manifest or not.

Extension commands by definition are excepted from this because if we can't parse the manifest, we don't know anything about extensions, since that's where they come from.

Reply:
sorry, re-reading my comment I see it might be unclear.

What I was referring to was the message in L317:
```
`To silence this warning, upgrade west.'
```
where I'm unsure how upgrading `west` would help ?

Reply:
> where I'm unsure how upgrading `west` would help ?

As far as I can tell, that message is only ever printed when the specific error is an instance of ManifestVersionError.

Am I missing a code path where it can be printed in some other situation? If not, upgrading west to the latest version is definitely a good fix to suggest to the user, since the latest west by definition will support the latest manifest version.

For context, since you weren't around when this work was done, ManifestVersionError was added as part of closing https://github.com/zephyrproject-rtos/west/issues/126 (which you opened) in https://github.com/zephyrproject-rtos/west/pull/324 (which I submitted).